core/trino-main/src/main/java/io/trino/sql/planner/planprinter/IoPlanPrinter.java

```diff
@@ -697,13 +754,14 @@ private void addInputTableConstraints(TupleDomain<ColumnHandle> filterDomain, Ta
             TableMetadata tableMetadata = plannerContext.getMetadata().getTableMetadata(session, table);
             TupleDomain<ColumnHandle> predicateDomain = plannerContext.getMetadata().getTableProperties(session, table).getPredicate();
             EstimatedStatsAndCost estimatedStatsAndCost = getEstimatedStatsAndCost(tableScan);
+            boolean withoutData = estimatedStatsAndCost.getOutputRowCount() == 0d ? true : false;
```

Comment:
The logic here should not depend on stats.
Stats are estimates, can be inaccurate and off.